### PR TITLE
add consts for hashes and test hash of argon2d

### PR DESCRIPTION
**Pull request by janriemer**
We define the resulting hashes for argon2d and
argon2i in a central place and add further
asserts to the generation of the hashes.

### CODE REVIEW

File: test.spec.js [hunk 1]
```diff
@@ -25,10 +30,12 @@ module.exports = {
...
     'use strict';
 
-    assert.expect(1);
+    assert.expect(3);
 
     argon2.hash(password, salt).then((hash) => {
-      assert.equal(hash, '$argon2i$m=4096,t=3,p=1$c29tZXNhbHQ$vpOd0mbc3AzXEHMgcTb1CrZt5XuoRQuz1kQtGBv7ejk');
+      assert.ok(hash, 'Hash should be defined.');
+      assert.ok(/\$argon2i\$/.test(hash), 'Should have argon2i signature.');
+      assert.equal(hash, hashes.argon2i);
```
Comment (ranisalt) [inline test.spec.js]: The test was like that before but that's very much redundant. If the hash is equal to the expected sample, then it certainly will have the `$argon2i$` signature and absolutely will be defined, as `undefined` returns `false` for every regex test.

**Comment by janriemer** [comment]
Ok, I've updated it.

**Comment by ranisalt** [comment]
Please try to rebase against master, else I can do it myself.

**Comment by janriemer** [comment]
Ok, you should be able to merge it now. :wink:

**Comment by ranisalt** [comment]
Travis is being utterly slow today, I guess their infrastructure is failing, but I'll merge anyway since the PR is simple and should not break.

**Pull request merged**